tests/common/reboot.py

```diff
@@ -79,9 +80,9 @@ def execute_reboot():
 
     if reboot_type == 'warm':
         logger.info('waiting for warmboot-finalizer service to finish')
-        finalizer_state = 'active'
+        finalizer_state = 'activating'
         count = 0
-        while finalizer_state == 'active':
+        while finalizer_state == 'activating':
```

Comment:
we need to make sure the systerm is loaded and finished. can you check that? if you run this check even before the server start, then it will not be in activating state and it will exit.

Reply:
@lguohan , 
- After warm-reboot  , the script waits for 120 seconds  before checking this warmboot-finalizer state. 
- All Checking critical services are verified in [mgmtVrf script ](https://github.com/Azure/sonic-mgmt/pull/1211/files#diff-22a048b57247f0f08cdf05039ec7fe7dR184)for all reboot services , once reboot is done . 
- Do you want to include [critical services check](https://github.com/Azure/sonic-mgmt/blob/master/tests/common/devices.py#L154) before checking warmboot-finalizer service state. ?

Reply:
what is the finishing state of warmboot finalizer service state? it is one-shot service, the final state should be dead?

Reply:
@lguohan  , the final state of warmboot finalizer service  is in-active(dead).
This is one shot service after warm-reboot is triggered , and once , warm-reboot is done , the service goes to in-active . 

After warm-reboot initiated  : 

```
root@MAA-Z9264F-9251:~#
● warmboot-finalizer.service - Monitor warm recovery and disable warmboot when d
   Loaded: loaded (/etc/systemd/system/warmboot-finalizer.service; disabled; ven
   Active: activating (start) since Sun 2016-12-18 13:24:17 UTC; 1min 56s ago
 Main PID: 1471 (finalize-warmbo)
    Tasks: 2 (limit: 4915)
   Memory: 1.3M
      CPU: 5.890s
   CGroup: /system.slice/warmboot-finalizer.service
           ├─ 1471 /bin/bash /usr/local/bin/finalize-warmboot.sh
           └─14602 sleep 5
```

After warm-reboot is  completed  : 

```
root@MAA-Z9264F-9251:/usr/bin# systemctl status warmboot-finalizer
● warmboot-finalizer.service - Monitor warm recovery and disable warmboot when done
   Loaded: loaded (/etc/systemd/system/warmboot-finalizer.service; disabled; vendor preset: enabled)
   Active: inactive (dead) since Sun 2016-12-18 13:26:56 UTC; 6min ago
  Process: 1471 ExecStart=/usr/local/bin/finalize-warmboot.sh (code=exited, status=0/SUCCESS)
 Main PID: 1471 (code=exited, status=0/SUCCESS)
      CPU: 9.254s

Dec 18 13:24:17 MAA-Z9264F-9251 systemd[1]: Starting Monitor warm recovery and disable warmboot when done...
Dec 18 13:26:50 MAA-Z9264F-9251 root[16230]: WARMBOOT_FINALIZER : Tearing down control plane assistant ...
Dec 18 13:26:50 MAA-Z9264F-9251 neighbor_advertiser[16235]: EVERFLOWV6 table does not exist
Dec 18 13:26:53 MAA-Z9264F-9251 neighbor_advertiser[16235]: Finish resetting mirror tunnel
Dec 18 13:26:53 MAA-Z9264F-9251 neighbor_advertiser[16235]: Finish resetting vxlan tunnel
Dec 18 13:26:54 MAA-Z9264F-9251 finalize-warmboot.sh[1471]: Running command: /usr/local/bin/sonic-cfggen -d --print-data > /etc/sonic/config_db.json
Dec 18 13:26:56 MAA-Z9264F-9251 sudo[16254]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/usr/bin/config warm_restart disable
Dec 18 13:26:56 MAA-Z9264F-9251 sudo[16254]: pam_unix(sudo:session): session opened for user root by (uid=0)
Dec 18 13:26:56 MAA-Z9264F-9251 systemd[1]: Started Monitor warm recovery and disable warmboot when done.
root@MAA-Z9264F-9251:/usr/bin#
```

Reply:
what is the initial state of the warmboot-finalizer? before activating state?

Reply:
The initial state is inactive.

```
root@sonic-s6100-07:~#  systemctl status warmboot-finalizer
● warmboot-finalizer.service - Monitor warm recovery and disable warmboot when done
   Loaded: loaded (/etc/systemd/system/warmboot-finalizer.service; disabled; vendor preset: enabled)
   Active: inactive (dead) since Wed 2020-01-29 09:30:20 UTC; 5min ago
  Process: 1903 ExecStart=/usr/local/bin/finalize-warmboot.sh (code=exited, status=0/SUCCESS)
 Main PID: 1903 (code=exited, status=0/SUCCESS)
      CPU: 355ms

Jan 29 09:30:19 sonic-s6100-07 systemd[1]: Starting Monitor warm recovery and disable warmboot when done...
Jan 29 09:30:19 sonic-s6100-07 root[1906]: WARMBOOT_FINALIZER : Wait for database to become ready...
Jan 29 09:30:20 sonic-s6100-07 root[2182]: WARMBOOT_FINALIZER : Database is ready...
Jan 29 09:30:20 sonic-s6100-07 systemd[1]: Started Monitor warm recovery and disable warmboot when done.
```

Reply:
in this case, if it is inactive, you need to wait till it becomes activating, and then active.

Reply:
Addressed